SkietbaanBE/SkietbaanBE/Migrations/ModelsContextModelSnapshot.cs

```diff
@@ -57,6 +93,81 @@ protected override void BuildModel(ModelBuilder modelBuilder)
 
                     b.ToTable("Users");
                 });
+
+            modelBuilder.Entity("SkietbaanBE.Models.UserCompStats", b =>
+                {
+                    b.Property<int>("Id")
+                        .ValueGeneratedOnAdd();
+
+                    b.Property<int>("BestScore");
+
+                    b.Property<int>("CompScore");
+
+                    b.Property<int?>("CompetitionId");
+
+                    b.Property<int?>("UserId");
+
+                    b.Property<DateTime>("Year");
+
+                    b.HasKey("Id");
+
+                    b.HasIndex("CompetitionId");
+
+                    b.HasIndex("UserId");
+
+                    b.ToTable("UserCompStats");
+                });
+
+            modelBuilder.Entity("SkietbaanBE.Models.UserGroup", b =>
+                {
+                    b.Property<int>("Id")
+                        .ValueGeneratedOnAdd();
+
+                    b.Property<int?>("GroupId");
+
+                    b.Property<int?>("UserId");
+
+                    b.HasKey("Id");
+
+                    b.HasIndex("GroupId");
+
+                    b.HasIndex("UserId");
+
+                    b.ToTable("UserGroups");
+                });
+
+            modelBuilder.Entity("SkietbaanBE.Models.Score", b =>
+                {
+                    b.HasOne("SkietbaanBE.Models.Competition", "Competition")
+                        .WithMany()
+                        .HasForeignKey("CompetitionId");
+
+                    b.HasOne("SkietbaanBE.Models.User", "User")
+                        .WithMany()
```

Comment:
please reference this properly 
`  modelBuilder.Entity<Bot>(entity =>
            {
                entity.HasOne(e => e.Client)
                    .WithMany(c => c.Bot)
                    .HasForeignKey(e => e.ClientId)
                    .OnDelete(DeleteBehavior.Cascade);
            });`

also within your model that is the one in the one to many put an icollection to indicate the many relationship
` public ICollection<Score> Score{ get; set; }` in your Competition class for example

Reply:
also in the many class just add a reference to your one class ex.
`public Competition Competition { get; set; }`

Reply:
I have changed the models and included the ICollection but I was advised not to change the auto generated files

Reply:
this means the models you generated off of was not entirely correct. this might definetely cause issues for you as soon as you try and manipulate these things through your context

Reply:
and do this for all of your replationships